flow/compositor_context.h

```diff
@@ -91,15 +91,20 @@ class FrameDamage {
 
   // See Damage::buffer_damage.
   std::optional<SkIRect> GetBufferDamage() {
-    return damage_ ? std::make_optional(damage_->buffer_damage) : std::nullopt;
+    return (damage_ && !ignore_damage_)
+               ? std::make_optional(damage_->buffer_damage)
+               : std::nullopt;
   }
 
+  void Reset() { ignore_damage_ = true; }
```

Comment:
I'm wondering if this could simply be `void Reset() { damage_ = std::nullopt; }`

Reply:
Interestingly if I do this rendering breaks. I wonder if the logic for computing damage requires all frames to have up to date damage calculations? Or if perhaps the std::nullopt signal is incorrect for frame damage and it should instead be treated as if the entire screen were dirty?

Reply:
I see. It breaks because in that case `frame_damage` is not properly reported in submit info which breaks tracking of accumulated damage in other two buffers. So the area subsequently repainted in back buffers is smaller than it should be be causing glitches. This is visible in the material progress indicator example when sliding in/out the menu:

https://user-images.githubusercontent.com/96958/234208847-4dd190e9-ede0-44c1-b167-0f370d2929dc.mov

Possible solution would be something like this https://github.com/knopp/engine/commit/30c3d61e960e35b2d3028944bf152991e4a68761 . Except *this doesn't work* because it causes `ShouldPerformPartialRepaint` always return `false` in subsequent repaints effectively disabling partial repaint completely.

So I'd keep it as it is, maybe add a comment that we need to preserve `frame_damage` to correctly track accumulated damage for other buffers while at same time reset `buffer_damage` to disable partial repaint for current buffer.

Reply:
Makes sense, thanks for the investigation here.  I added a comment, but I think it makes sense to keep frame damage because the damage info is still correct a subsequent frame could use this and decide to do a partial repaint.